src/satosa/yaml.py

```diff
@@ -0,0 +1,58 @@
+import os
+import re
+
+from yaml import SafeLoader as _safe_loader
+from yaml import YAMLError
+from yaml import safe_load as load
+
+
+def _constructor_env_variables(loader, node):
+    """
+    Extracts the environment variable from the node's value.
+    :param yaml.Loader loader: the yaml loader
+    :param node: the current node in the yaml
+    :return: value of the environment variable
+    """
+    raw_value = loader.construct_scalar(node)
+    new_value = os.environ.get(raw_value)
+    if new_value is None:
+        msg = "Cannot construct value from {node}: {value}".format(
+            node=node, value=new_value
+        )
+        raise YAMLError(msg)
+    return new_value
+
+
+def _constructor_envfile_variables(loader, node):
+    """
+    Extracts the environment variable from the node's value.
+    :param yaml.Loader loader: the yaml loader
+    :param node: the current node in the yaml
+    :return: value read from file pointed to by environment variable
+    """
+    raw_value = loader.construct_scalar(node)
+    filepath = os.environ.get(raw_value)
+    if filepath is None:
+        msg = "Cannot construct value from {node}: {path}".format(
+            node=node, path=filepath
+        )
+        raise YAMLError(msg)
+
+    try:
+        with open(filepath, "r") as fd:
+            new_value = fd.read()
```

Comment:
`.strip()` doesn't take place anymore.

Reply:
Can we assume that everyone will want the contents of a file to be `.strip()`ed?

_Not_ doing this, means that the user can later choose to do so, later; it also means that the user is responsible for the file contents, and that content should be properly catered if no such whitespace is wanted.
Doing this here, means that the user has no chance to restore the stripped characters.